src/core/src/FabricResourceManager.cpp

```diff
@@ -91,7 +91,10 @@ std::shared_ptr<FabricGeometry> FabricResourceManager::acquireGeometry(
 }
 
 bool useSharedMaterial(const FabricMaterialDefinition& materialDefinition) {
-    if (materialDefinition.hasBaseColorTexture() || materialDefinition.getImageryLayerCount() > 0) {
+    if (materialDefinition.hasBaseColorTexture() || materialDefinition.getImageryLayerCount() > 0 ||
+        !materialDefinition.getFeatureIdTypes().empty() ||
+        !materialDefinition.getMdlInternalPropertyAttributePropertyTypes().empty() ||
+        !materialDefinition.getMdlInternalPropertyTexturePropertyTypes().empty()) {
```

Comment:
Bug fix: can't share materials if there are things that are unique to that material like feature ids and properties